### ISSUE TITLE
Rotating a MultiPolygonAdapter produces an invalid `geoms` attribute

### ISSUE THREAD

**Issue by jacopofar** [Toblerity/Shapely#1100]
Hello, and thanks for the useful library!

I found out that when I rotate a MultiPolygon the resulting object doesn't allow access to the `geoms` attribute, making it impossible to draw it using Matplotlib or convert to a geoJSON.

I see the rotation code mentions a [problem with the GeometryCollection constructor](https://github.com/Toblerity/Shapely/blob/eae4dcbedb0756f658f9982825d5fa23b0061b2a/shapely/affinity.py#L91) but I don't what that problem is and if it's related.

## Expected behavior and actual behavior.

After the rotation I can access the `geoms` field of the rotated object, instead get a `TypeError: 'Polygon' object is not subscriptable`:

    >>> list(rotated.geoms)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/usr/local/lib/python3.7/dist-packages/shapely/geometry/base.py", line 966, in __iter__
        self._update()
      File "/usr/local/lib/python3.7/dist-packages/shapely/geometry/base.py", line 955, in _update
        self._ndim = self.__p__._ndim
      File "/usr/local/lib/python3.7/dist-packages/shapely/geometry/multipolygon.py", line 119, in _ndim
        array = self.context[0][0].__array_interface__
    TypeError: 'Polygon' object is not subscriptable

## Steps to reproduce the problem.

I wrote a minimal script to reproduce the problem
<details>
 <summary>Script</summary>

```python
  from shapely.geometry import shape, mapping, asShape
  from shapely import affinity
  
  gj = {
      "type": "MultiPolygon",
      "coordinates": [
          [
              [
                  [1336988.5732014342, 7189788.468580844],
                  [1336989.3524378699, 7189773.279752336],
                  [1336994.8070929185, 7189765.875209061],
                  [1337008.4325985918, 7189765.495489079],
                  [1337019.3419086896, 7189771.722899069],
                  [1337028.681613967, 7189782.222171575],
                  [1337029.8504686204, 7189797.031295827],
                  [1337021.2788678291, 7189804.416882307],
                  [1337000.2617479675, 7189805.195312136],
                  [1336997.1448022253, 7189802.081593289],
                  [1336988.5732014342, 7189788.468580844],
              ]
          ]
      ],
  }
  original = asShape(gj)
  assert original.type == "MultiPolygon"
  
  rotated = affinity.rotate(original, 90, origin="centroid")
  
  assert rotated.type == "MultiPolygon"

  list(original.geoms)
  
  restored = mapping(rotated)
  list(rotated.geoms)
```
</details>

The object comes straight from PostGIS, and it's a representation of [this OSM way](https://www.openstreetmap.org/way/89545913)

## Operating system
I could reproduce it in two environments:

* macOS 11.2 running Python 3.9 and Shapely 1.7.1 
* Ubuntu running Python 3.7 and Shapely 1.7.1 

## Shapely version and provenance

Shapely 1.7.1  installed using pip

**Comment by jorisvandenbossche** [comment]
In the reproducible script, you are using `asShape` to create the MultiPolygon. This function actually creates a "proxy" object, and not an actual MultiPolygon geometry. Those proxy objects will be deprecated in the upcoming Shapely 1.8 release, and you can use `shape` instead to convert a geojson-like mapping to a geometry object. 

And if replacing `asShape` with `shape` in the script above, this actually works correctly. So it seems the bug is specific to the deprecated proxy classes (and thus IMO also not a priority to fix for 1.8).

**Comment by jacopofar** [comment]
Thanks, I just tried it and with `shape` it works on Shapely 1.7.1.

Here's the code with that change, when running doesn't raise exceptions anymore:
<details>

```python
from shapely.geometry import shape, mapping, asShape
from shapely import affinity

gj = {
  "type": "MultiPolygon",
  "coordinates": [
      [
          [
              [1336988.5732014342, 7189788.468580844],
              [1336989.3524378699, 7189773.279752336],
              [1336994.8070929185, 7189765.875209061],
              [1337008.4325985918, 7189765.495489079],
              [1337019.3419086896, 7189771.722899069],
              [1337028.681613967, 7189782.222171575],
              [1337029.8504686204, 7189797.031295827],
              [1337021.2788678291, 7189804.416882307],
              [1337000.2617479675, 7189805.195312136],
              [1336997.1448022253, 7189802.081593289],
              [1336988.5732014342, 7189788.468580844],
          ]
      ]
  ],
}
original = shape(gj)
assert original.type == "MultiPolygon"

rotated = affinity.rotate(original, 90, origin="centroid")

assert rotated.type == "MultiPolygon"

list(original.geoms)

restored = mapping(rotated)
list(rotated.geoms)

```
</details>
